crates/proof-of-sql/src/base/database/table.rs

```diff
@@ -18,18 +23,20 @@ pub enum TableError {
 #[derive(Debug, Clone, Eq)]
 pub struct Table<'a, S: Scalar> {
     table: IndexMap<Identifier, Column<'a, S>>,
+    num_rows: usize,
 }
 impl<'a, S: Scalar> Table<'a, S> {
     /// Creates a new [`Table`].
     pub fn try_new(table: IndexMap<Identifier, Column<'a, S>>) -> Result<Self, TableError> {
         if table.is_empty() {
-            return Ok(Self { table });
+            // `EmptyExec` should have one row for queries such as `SELECT 1`.
+            return Ok(Self { table, num_rows: 1 });
```

Comment:
This default behavior doesn't really make sense.
This is what arrow does:
https://github.com/apache/arrow-rs/blob/3ee5048c8ea3aa531d111afe33d0a3551eabcd84/arrow-array/src/record_batch.rs#L86
https://github.com/apache/arrow-rs/blob/3ee5048c8ea3aa531d111afe33d0a3551eabcd84/arrow-array/src/record_batch.rs#L501